api/v1beta2/ibmpowervsmachine_types.go

```diff
@@ -146,6 +146,8 @@ type IBMPowerVSResourceReference struct {
 
 	// Name of resource
 	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength:=63
```

Comment:
@Shilpa-Gokul, can you share why this validation was added to `IBMPowerVSResourceReference`? `ResourceGroup` is of type `IBMPowerVSResourceReference` and I see that is supports camel case and capital letters in case user specifies it.
Would it maybe be better to add to the specific resource than adding it here?